Sources/RInAppMessaging/Router.swift

```diff
@@ -62,65 +62,71 @@ internal class Router: RouterType {
         }
 
         displayQueue.async {
-
-            func getPresenter<T>(type: T.Type) -> T? {
-                guard let presenter = self.dependencyManager.resolve(type: type) else {
-                    Logger.debug("Error: \(type) couldn't be resolved")
-                    return nil
-                }
-                return presenter
-            }
-
-            var viewConstructor: (() -> BaseView)?
-            switch campaignViewType {
-            case .modal:
-                guard let presenter = getPresenter(type: FullViewPresenterType.self) else {
-                    break
-                }
-                presenter.campaign = campaign
-                if let associatedImageData = associatedImageData {
-                    presenter.associatedImage = UIImage(data: associatedImageData)
-                }
-                viewConstructor = { ModalView(presenter: presenter) }
-            case .full:
-                guard let presenter = getPresenter(type: FullViewPresenterType.self) else {
-                    break
-                }
-                presenter.campaign = campaign
-                if let associatedImageData = associatedImageData {
-                    presenter.associatedImage = UIImage(data: associatedImageData)
-                }
-                viewConstructor = { FullScreenView(presenter: presenter) }
-            case .slide:
-                guard let presenter = getPresenter(type: SlideUpViewPresenterType.self) else {
-                    break
-                }
-                presenter.campaign = campaign
-                viewConstructor = { SlideUpView(presenter: presenter) }
-            case .invalid, .html:
-                Logger.debug("Error: Campaign view type not supported")
+            guard let viewConstructor = self.viewConstructor(for: campaign, associatedImageData: associatedImageData) else {
+                completion(true)
+                return
             }
 
             DispatchQueue.main.async {
-                guard let view = viewConstructor?(),
-                      let rootView = UIApplication.shared.getKeyWindow(),
+                guard let rootView = UIApplication.shared.getKeyWindow(),
                       rootView.findIAMViewSubview() == nil,
-                      confirmation() == true else {
+                      confirmation() else {
 
                     completion(true)
                     return
                 }
 
+                let view = viewConstructor()
```

Comment:
viewConstructor returns an optional function - i.e. can be nil - but view is accessed directly below `view.show()` and you didn't need to unwrap it. Why is that?

Reply:
viewConstructor is unwrapped in guard a few lines above. And when executed it returns `BaseView` as non-optional.
It's probably confusing because `viewConstructor(Campaign, Data?)` returns a closure that is assigned to `viewConstructor` local variable. I'll change the naming